### PR TITLE
params: read from ParamsDependency if possible

### LINKED ISSUE #5451: params: untracked values in params files are not ignored in params/exp commands
# Bug Report

exp show: Invalid or wrong values for column names and values

## Description

I have an experiment trying different parameters, it is the last stage of the pipeline, and the only with checkpointed outputs. Following this procedure: run the experiment (`dvc exp run`), change a parameter in `params.yaml` and run it again, do it several times. To visualize the results I run `dvc exp show`, but I get wrong column names and values.

I have a doubt about the intended behavior of this feature, when `showing` experiments, the idea is to show the comparison of metrics along parameters. It is expected to show the difference in metrics for ALL stages that contain metrics and parameters, or only those being experimented with (the ones that are run when iterating experiments)?

Trying to hide the trash columns with `--exclude-metrics` allows me to remove some columns, though some others are not recognized. But, in any case, it doesn't remove the strange field values. I include a photo to avoid breaking the format:

2021-02-11-171417_1895x84_scrot

### Reproduce

Reproducing is a bit computationally expensive, it requires training a bert model and doing some operations on it's embeddings, it can be accomplished in the following way (at least one GPU with enough RAM, 5-8hrs of computations).

```console
mkdir ./venv
virtualenv ./venv/dvc-reproduce
source ./venv/dvc-reproduce/bin/activate
git clone https://github.com/geblanco/quail-experiments
cd quail-experiments
git checkout restrucutre-experiments
mkdir ~/dvc_cache
dvc config --local core.remote local_dvc
dvc remote add --local -d local_dvc "~/dvc_cache"
dvc repro merge_norm_with_lengths_oversample_05
dvc exp run
```

### Expected

The command should show a table with the parameters tracked by the stage, it's changes and results in metrics.

### Environment information

**Output of `dvc version`:**

```console
$ dvc version
2.0.0a0+02cfe3
```

**Additional Information (if any):**

I have the intuition that it has to do with the parsing of parameters from a file other than `params.yaml`. To be more clear, here is the output in json of the `show` command:

There are various entries for `data/specs/bert-train-no-empty-answers_experiment.json` stated as parameters, with the whole file contents as parameters, although the stage using that file only uses a part of it as parameters (see stage bert-train-no-empty-answers in `dvc.yaml` below)

```console
dvc exp show --show-json
```
```json
{                                                                     
    "workspace": {
        "baseline": {
            "timestamp": null,
            "params": {
                "data/specs/bert-train-no-empty-answers_experiment.json": {
                    "inputs": [
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/train.json",
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/dev.json",
                        "/home/gb/Documents/Research/quail-experiments/data/models/bert-base-uncased"
                    ],
                    "scripts": [
                        "/home/gb/Documents/Research/quail-experiments/src/processing/run.sh"
                    ],
                    "metrics": [
                        "train_metrics.json",
                        "eval_metrics.json"
                    ],
                    "outputs": [
                        "data/models/bert-train-no-empty-answers/config.json",
                        "data/models/bert-train-no-empty-answers/pytorch_model.bin",
                        "data/models/bert-train-no-empty-answers/special_tokens_map.json",
                        "data/models/bert-train-no-empty-answers/tokenizer_config.json",
                        "data/models/bert-train-no-empty-answers/training_args.bin",
                        "data/models/bert-train-no-empty-answers/vocab.txt"
                    ],
                    "results": [
                        "train_predictions.json",
                        "train_nbest_predictions.json",
                        "eval_predictions.json",
                        "eval_nbest_predictions.json"
                    ],
                    "command": [
                        "./src/processing/run.sh",
                        "data/specs/bert-train-no-empty-answers_experiment.json"
                    ],
                    "params": {
                        "meta": "bert-train-no-empty-answers",
                        "data_dir": "data/quail_no_empty_answers",
                        "cache_dir": "/tmp",
                        "model_name_or_path": "data/models/bert-base-uncased",
                        "output_dir": "data/models/bert-train-no-empty-answers",
                        "metrics_dir": "data/metrics/bert-train-no-empty-answers",
                        "results_dir": "data/results/bert-train-no-empty-answers",
                        "model_type": "bert",
                        "task_name": "generic",
                        "do_train": true,
                        "do_eval": true,
                        "fp16": true,
                        "fp16_opt_level": "O1",
                        "save_total_limit": 0,
                        "save_steps": 0,
                        "max_seq_length": 484,
                        "num_train_epochs": 2,
                        "per_device_eval_batch_size": 8,
                        "per_device_train_batch_size": 2,
                        "gradient_accumulation_steps": 8,
                        "learning_rate": 5e-05,
                        "warmup_steps": 500,
                        "save_logits": true
                    },
                    "qualified_name": "bert-train-no-empty-answers",
                    "experiment_dir": "bert-train-no-empty-answers",
                    "experiment_name": "bert-train-no-empty-answers"
                },
                "params.yaml": {
                    "classification": {
                        "iterations": 200,
                        "popsize": 50,
                        "selection": 10,
                        "early_stop": 6,
                        "balanced": false,
                        "memory": 64,
                        "autogoal": false,
                        "pipeline": "logreg",
                        "sweep_features": false,
                        "test_size": 0.2,
                        "metric": "weighted_f1",
                        "multi_layer": {
                            "lr": 0.001,
                            "epochs": 100,
                            "batch_size": 200
                        }
                    },
                    "features": {
                        "normalization": true,
                        "oversample": false,
                        "text_length": false,
                        "embeddings": false,
                        "logits": false,
                        "context": false,
                        "question": false,
                        "endings": false
                    }
                }
            },
            "queued": false,
            "metrics": {
                "data/metrics/bert-train-no-empty-answers/eval_metrics.json": {
                    "eval_loss": 1.0892217214358713,
                    "eval_acc": 0.5046210720887245
                },
                "data/metrics/bert-train-no-empty-answers/train_metrics.json": {
                    "eval_loss": 0.5457088775723894,
                    "eval_acc": 0.8294667399670148
                },
                "data/metrics/classification/scores.json": {
                    "0": {
                        "precision": 0.0,
                        "recall": 0.0,
                        "f1-score": 0.0,
                        "support": 315
                    },
                    "1": {
                        "precision": 0.8268279274326553,
                        "recall": 1.0,
                        "f1-score": 0.9052061390309961,
                        "support": 1504
                    },
                    "accuracy": 0.8268279274326553,
                    "macro avg": {
                        "precision": 0.41341396371632766,
                        "recall": 0.5,
                        "f1-score": 0.45260306951549806,
                        "support": 1819
                    },
                    "weighted avg": {
                        "precision": 0.6836444215825803,
                        "recall": 0.8268279274326553,
                        "f1-score": 0.7484497158343145,
                        "support": 1819
                    }
                }
            }
        }
    },
    "1a64cf41824def258db67785f8b249d7c95cd46c": {
        "baseline": {
            "timestamp": "2021-02-10T17:01:56",
            "params": {
                "data/specs/bert-train-no-empty-answers_experiment.json": {
                    "inputs": [
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/train.json",
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/dev.json",
                        "/home/gb/Documents/Research/quail-experiments/data/models/bert-base-uncased"
                    ],
                    "scripts": [
                        "/home/gb/Documents/Research/quail-experiments/src/processing/run.sh"
                    ],
                    "metrics": [
                        "train_metrics.json",
                        "eval_metrics.json"
                    ],
                    "outputs": [
                        "data/models/bert-train-no-empty-answers/config.json",
                        "data/models/bert-train-no-empty-answers/pytorch_model.bin",
                        "data/models/bert-train-no-empty-answers/special_tokens_map.json",
                        "data/models/bert-train-no-empty-answers/tokenizer_config.json",
                        "data/models/bert-train-no-empty-answers/training_args.bin",
                        "data/models/bert-train-no-empty-answers/vocab.txt"
                    ],
                    "results": [
                        "train_predictions.json",
                        "train_nbest_predictions.json",
                        "eval_predictions.json",
                        "eval_nbest_predictions.json"
                    ],
                    "command": [
                        "./src/processing/run.sh",
                        "data/specs/bert-train-no-empty-answers_experiment.json"
                    ],
                    "params": {
                        "meta": "bert-train-no-empty-answers",
                        "data_dir": "data/quail_no_empty_answers",
                        "cache_dir": "/tmp",
                        "model_name_or_path": "data/models/bert-base-uncased",
                        "output_dir": "data/models/bert-train-no-empty-answers",
                        "metrics_dir": "data/metrics/bert-train-no-empty-answers",
                        "results_dir": "data/results/bert-train-no-empty-answers",
                        "model_type": "bert",
                        "task_name": "generic",
                        "do_train": true,
                        "do_eval": true,
                        "fp16": true,
                        "fp16_opt_level": "O1",
                        "save_total_limit": 0,
                        "save_steps": 0,
                        "max_seq_length": 484,
                        "num_train_epochs": 2,
                        "per_device_eval_batch_size": 8,
                        "per_device_train_batch_size": 2,
                        "gradient_accumulation_steps": 8,
                        "learning_rate": 5e-05,
                        "warmup_steps": 500,
                        "save_logits": true
                    },
                    "qualified_name": "bert-train-no-empty-answers",
                    "experiment_dir": "bert-train-no-empty-answers",
                    "experiment_name": "bert-train-no-empty-answers"
                },
                "params.yaml": {
                    "classification": {
                        "iterations": 200,
                        "popsize": 50,
                        "selection": 10,
                        "early_stop": 6,
                        "balanced": false,
                        "memory": 64,
                        "autogoal": false,
                        "pipeline": "logreg",
                        "sweep_features": false,
                        "test_size": 0.2,
                        "metric": "weighted_f1",
                        "multi_layer": {
                            "lr": 0.001,
                            "epochs": 100,
                            "batch_size": 200
                        }
                    },
                    "features": {
                        "normalization": false,
                        "oversample": false,
                        "text_length": false,
                        "embeddings": false,
                        "logits": false,
                        "context": false,
                        "question": false,
                        "endings": false
                    }
                }
            },
            "queued": false,
            "metrics": {
                "data/metrics/bert-train-no-empty-answers/eval_metrics.json": {
                    "eval_loss": 1.0892217214358713,
                    "eval_acc": 0.5046210720887245
                },
                "data/metrics/bert-train-no-empty-answers/train_metrics.json": {
                    "eval_loss": 0.5457088775723894,
                    "eval_acc": 0.8294667399670148
                }
            },
            "name": "restrucutre-experiments"
        },
        "7936e1fd371ceca8599b312930a27e3acdbadbe5": {
            "checkpoint_tip": "7936e1fd371ceca8599b312930a27e3acdbadbe5",
            "timestamp": "2021-02-11T11:47:46",
            "params": {
                "data/specs/bert-train-no-empty-answers_experiment.json": {
                    "inputs": [
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/train.json",
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/dev.json",
                        "/home/gb/Documents/Research/quail-experiments/data/models/bert-base-uncased"
                    ],
                    "scripts": [
                        "/home/gb/Documents/Research/quail-experiments/src/processing/run.sh"
                    ],
                    "metrics": [
                        "train_metrics.json",
                        "eval_metrics.json"
                    ],
                    "outputs": [
                        "data/models/bert-train-no-empty-answers/config.json",
                        "data/models/bert-train-no-empty-answers/pytorch_model.bin",
                        "data/models/bert-train-no-empty-answers/special_tokens_map.json",
                        "data/models/bert-train-no-empty-answers/tokenizer_config.json",
                        "data/models/bert-train-no-empty-answers/training_args.bin",
                        "data/models/bert-train-no-empty-answers/vocab.txt"
                    ],
                    "results": [
                        "train_predictions.json",
                        "train_nbest_predictions.json",
                        "eval_predictions.json",
                        "eval_nbest_predictions.json"
                    ],
                    "command": [
                        "./src/processing/run.sh",
                        "data/specs/bert-train-no-empty-answers_experiment.json"
                    ],
                    "params": {
                        "meta": "bert-train-no-empty-answers",
                        "data_dir": "data/quail_no_empty_answers",
                        "cache_dir": "/tmp",
                        "model_name_or_path": "data/models/bert-base-uncased",
                        "output_dir": "data/models/bert-train-no-empty-answers",
                        "metrics_dir": "data/metrics/bert-train-no-empty-answers",
                        "results_dir": "data/results/bert-train-no-empty-answers",
                        "model_type": "bert",
                        "task_name": "generic",
                        "do_train": true,
                        "do_eval": true,
                        "fp16": true,
                        "fp16_opt_level": "O1",
                        "save_total_limit": 0,
                        "save_steps": 0,
                        "max_seq_length": 484,
                        "num_train_epochs": 2,
                        "per_device_eval_batch_size": 8,
                        "per_device_train_batch_size": 2,
                        "gradient_accumulation_steps": 8,
                        "learning_rate": 5e-05,
                        "warmup_steps": 500,
                        "save_logits": true
                    },
                    "qualified_name": "bert-train-no-empty-answers",
                    "experiment_dir": "bert-train-no-empty-answers",
                    "experiment_name": "bert-train-no-empty-answers"
                },
                "params.yaml": {
                    "classification": {
                        "iterations": 200,
                        "popsize": 50,
                        "selection": 10,
                        "early_stop": 6,
                        "balanced": false,
                        "memory": 64,
                        "autogoal": false,
                        "pipeline": "logreg",
                        "sweep_features": false,
                        "test_size": 0.2,
                        "metric": "weighted_f1",
                        "multi_layer": {
                            "lr": 0.001,
                            "epochs": 100,
                            "batch_size": 200
                        }
                    },
                    "features": {
                        "normalization": true,
                        "oversample": false,
                        "text_length": false,
                        "embeddings": false,
                        "logits": false,
                        "context": false,
                        "question": false,
                        "endings": false
                    }
                }
            },
            "queued": false,
            "metrics": {
                "data/metrics/bert-train-no-empty-answers/eval_metrics.json": {
                    "eval_loss": 1.0892217214358713,
                    "eval_acc": 0.5046210720887245
                },
                "data/metrics/bert-train-no-empty-answers/train_metrics.json": {
                    "eval_loss": 0.5457088775723894,
                    "eval_acc": 0.8294667399670148
                },
                "data/metrics/classification/scores.json": {
                    "0": {
                        "precision": 0.0,
                        "recall": 0.0,
                        "f1-score": 0.0,
                        "support": 315
                    },
                    "1": {
                        "precision": 0.8268279274326553,
                        "recall": 1.0,
                        "f1-score": 0.9052061390309961,
                        "support": 1504
                    },
                    "accuracy": 0.8268279274326553,
                    "macro avg": {
                        "precision": 0.41341396371632766,
                        "recall": 0.5,
                        "f1-score": 0.45260306951549806,
                        "support": 1819
                    },
                    "weighted avg": {
                        "precision": 0.6836444215825803,
                        "recall": 0.8268279274326553,
                        "f1-score": 0.7484497158343145,
                        "support": 1819
                    }
                }
            },
            "name": "exp-6b84d",
            "checkpoint_parent": "b5f88821b90096dd4ed14ca397e7c1fdd28fdd64"
        },
        "b5f88821b90096dd4ed14ca397e7c1fdd28fdd64": {
            "checkpoint_tip": "b5f88821b90096dd4ed14ca397e7c1fdd28fdd64",
            "checkpoint_parent": "335edd1f67f3908e70b872f307a2500cbd47c6ac"
        },
        "335edd1f67f3908e70b872f307a2500cbd47c6ac": {
            "checkpoint_tip": "335edd1f67f3908e70b872f307a2500cbd47c6ac",
            "checkpoint_parent": "e842154c2a8268bcba9da8bc5e9775b2e13d409e"
        },
        "e842154c2a8268bcba9da8bc5e9775b2e13d409e": {
            "checkpoint_tip": "e842154c2a8268bcba9da8bc5e9775b2e13d409e",
            "checkpoint_parent": "d1db1a587c83847812f47e9ba4e68dac95886350"
        },
        "d1db1a587c83847812f47e9ba4e68dac95886350": {
            "timestamp": "2021-02-10T18:07:58",
            "params": {
                "data/specs/bert-train-no-empty-answers_experiment.json": {
                    "inputs": [
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/train.json",
                        "/home/gb/Documents/Research/quail-experiments/data/quail_no_empty_answers/dev.json",
                        "/home/gb/Documents/Research/quail-experiments/data/models/bert-base-uncased"
                    ],
                    "scripts": [
                        "/home/gb/Documents/Research/quail-experiments/src/processing/run.sh"
                    ],
                    "metrics": [
                        "train_metrics.json",
                        "eval_metrics.json"
                    ],
                    "outputs": [
                        "data/models/bert-train-no-empty-answers/config.json",
                        "data/models/bert-train-no-empty-answers/pytorch_model.bin",
                        "data/models/bert-train-no-empty-answers/special_tokens_map.json",
                        "data/models/bert-train-no-empty-answers/tokenizer_config.json",
                        "data/models/bert-train-no-empty-answers/training_args.bin",
                        "data/models/bert-train-no-empty-answers/vocab.txt"
                    ],
                    "results": [
                        "train_predictions.json",
                        "train_nbest_predictions.json",
                        "eval_predictions.json",
                        "eval_nbest_predictions.json"
                    ],
                    "command": [
                        "./src/processing/run.sh",
                        "data/specs/bert-train-no-empty-answers_experiment.json"
                    ],
                    "params": {
                        "meta": "bert-train-no-empty-answers",
                        "data_dir": "data/quail_no_empty_answers",
                        "cache_dir": "/tmp",
                        "model_name_or_path": "data/models/bert-base-uncased",
                        "output_dir": "data/models/bert-train-no-empty-answers",
                        "metrics_dir": "data/metrics/bert-train-no-empty-answers",
                        "results_dir": "data/results/bert-train-no-empty-answers",
                        "model_type": "bert",
                        "task_name": "generic",
                        "do_train": true,
                        "do_eval": true,
                        "fp16": true,
                        "fp16_opt_level": "O1",
                        "save_total_limit": 0,
                        "save_steps": 0,
                        "max_seq_length": 484,
                        "num_train_epochs": 2,
                        "per_device_eval_batch_size": 8,
                        "per_device_train_batch_size": 2,
                        "gradient_accumulation_steps": 8,
                        "learning_rate": 5e-05,
                        "warmup_steps": 500,
                        "save_logits": true
                    },
                    "qualified_name": "bert-train-no-empty-answers",
                    "experiment_dir": "bert-train-no-empty-answers",
                    "experiment_name": "bert-train-no-empty-answers"
                },
                "params.yaml": {
                    "classification": {
                        "iterations": 200,
                        "popsize": 50,
                        "selection": 10,
                        "early_stop": 6,
                        "balanced": false,
                        "memory": 64,
                        "autogoal": false,
                        "pipeline": "logreg",
                        "sweep_features": false,
                        "test_size": 0.2,
                        "metric": "weighted_f1",
                        "multi_layer": {
                            "lr": 0.001,
                            "epochs": 100,
                            "batch_size": 200
                        }
                    },
                    "features": {
                        "normalization": true,
                        "oversample": false,
                        "text_length": false,
                        "embeddings": false,
                        "logits": false,
                        "context": false,
                        "question": false,
                        "endings": false
                    }
                }
            },
            "queued": false,
            "metrics": {
                "data/metrics/bert-train-no-empty-answers/eval_metrics.json": {
                    "eval_loss": 1.0892217214358713,
                    "eval_acc": 0.5046210720887245
                },
                "data/metrics/bert-train-no-empty-answers/train_metrics.json": {
                    "eval_loss": 0.5457088775723894,
                    "eval_acc": 0.8294667399670148
                },
                "data/metrics/classification/scores.json": {
                    "0": {
                        "precision": 0.0,
                        "recall": 0.0,
                        "f1-score": 0.0,
                        "support": 329
                    },
                    "1": {
                        "precision": 0.8191313908741067,
                        "recall": 1.0,
                        "f1-score": 0.9005741915986704,
                        "support": 1490
                    },
                    "accuracy": 0.8191313908741067,
                    "macro avg": {
                        "precision": 0.40956569543705335,
                        "recall": 0.5,
                        "f1-score": 0.4502870957993352,
                        "support": 1819
                    },
                    "weighted avg": {
                        "precision": 0.6709762355153485,
                        "recall": 0.8191313908741067,
                        "f1-score": 0.7376885901495431,
                        "support": 1819
                    }
                }
            },
            "name": "exp-a1b47"
        }
    }
}
```

```yaml
stages:
  dl_quail:
    cmd: ./src/etl/dl_quail.sh
    deps:
    - src/etl/dl_quail.sh
    outs:
    - data/quail/dev.json
    - data/quail/dev_answers.json
    - data/quail/eval.py
    - data/quail/train.json
    - data/quail/train_answers.json
  dl_models:
    cmd: python ./src/etl/dl_models.py -m bert-base-uncased --overwrite
    deps:
    - src/etl/dl_models.py
    outs:
    - data/models/bert-base-uncased
  tr_quail_train_to_race:
    cmd: src/etl/preprocess_quail_to_race.py data/quail/train.json data/quail/train_answers.json
      -o data/quail_race_fmt/train.json
    deps:
    - data/quail/train.json
    - data/quail/train_answers.json
    - src/etl/preprocess_quail_to_race.py
    outs:
    - data/quail_race_fmt/train.json
  tr_quail_dev_to_race:
    cmd: src/etl/preprocess_quail_to_race.py data/quail/dev.json data/quail/dev_answers.json
      -o data/quail_race_fmt/dev.json
    deps:
    - data/quail/dev.json
    - data/quail/dev_answers.json
    - src/etl/preprocess_quail_to_race.py
    outs:
    - data/quail_race_fmt/dev.json
  tr_quail_train_no_empty_answers:
    cmd: python src/etl/choices/reformat_dataset.py -d data/quail_race_fmt -o data/quail_no_empty_answers
      -s train --no_answer_text "not enough information"
    deps:
    - data/quail_race_fmt
    - src/etl/choices/reformat_dataset.py
    outs:
    - data/quail_no_empty_answers/train.json
  tr_quail_dev_no_empty_answers:
    cmd: python src/etl/choices/reformat_dataset.py -d ./data/quail_race_fmt -o ./data/quail_no_empty_answers
      -s dev --no_answer_text "not enough information" --keep_matching_text --index_list_path
      ./data/quail_no_empty_answers/dev_index_list.json
    deps:
    - data/quail_race_fmt
    - src/etl/choices/reformat_dataset.py
    outs:
    - ./data/quail_no_empty_answers/dev_index_list.json
    - data/quail_no_empty_answers/dev.json
  bert-train-no-empty-answers:
    cmd: ./src/processing/run.sh data/specs/bert-train-no-empty-answers_experiment.json
    deps:
    - data/models/bert-base-uncased
    - data/quail_no_empty_answers/dev.json
    - data/quail_no_empty_answers/train.json
    - src/processing/run.sh
    params:
    - data/specs/bert-train-no-empty-answers_experiment.json:
      - params.do_eval
      - params.do_train
      - params.fp16
      - params.fp16_opt_level
      - params.gradient_accumulation_steps
      - params.learning_rate
      - params.max_seq_length
      - params.meta
      - params.model_type
      - params.num_train_epochs
      - params.per_device_eval_batch_size
      - params.per_device_train_batch_size
      - params.task_name
      - params.warmup_steps
    outs:
    - data/models/bert-train-no-empty-answers/config.json
    - data/models/bert-train-no-empty-answers/pytorch_model.bin
    - data/models/bert-train-no-empty-answers/special_tokens_map.json
    - data/models/bert-train-no-empty-answers/tokenizer_config.json
    - data/models/bert-train-no-empty-answers/training_args.bin
    - data/models/bert-train-no-empty-answers/vocab.txt
    - data/results/bert-train-no-empty-answers/eval_nbest_predictions.json
    - data/results/bert-train-no-empty-answers/eval_predictions.json
    - data/results/bert-train-no-empty-answers/train_nbest_predictions.json
    - data/results/bert-train-no-empty-answers/train_predictions.json
    metrics:
    - data/metrics/bert-train-no-empty-answers/eval_metrics.json
    - data/metrics/bert-train-no-empty-answers/train_metrics.json
  extract_no_empty_answers_embeddings:
    cmd: python src/etl/embeddings/extract/extract_embeddings.py -s train --args_file
      data/specs/bert-eval-no-empty-answers_experiment.json --extract params --scatter_dataset
      -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2 --oversample 0.5
    deps:
    - data/models/bert-train-no-empty-answers
    - data/quail_no_empty_answers/train.json
    - data/specs/bert-eval-no-empty-answers_experiment.json
    - src/etl/embeddings/extract/extract_embeddings.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/embeddings
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_embeddings
  extract_no_empty_answers_lengths:
    cmd: python src/etl/embeddings/extract/extract_lengths.py -d data/quail_no_empty_answers
      -s train -t generic -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths
    deps:
    - data/quail_no_empty_answers/train.json
    - src/etl/embeddings/extract/extract_lengths.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_data.pkl
  extract_no_empty_answers_lengths_oversample_05:
    cmd: python src/etl/embeddings/extract/extract_lengths.py -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_embeddings/
      -s train -t generic -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_oversample
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_embeddings/train.json
    - src/etl/embeddings/extract/extract_lengths.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_oversample_data.pkl
  normalize_no_emtpy_answers_embeddings:
    cmd: python src/etl/embeddings/transform/normalize_embeddings.py -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/embeddings
      -n 50 --scatter_dataset -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/embeddings
    - src/etl/embeddings/transform/normalize_embeddings.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings
  normalize_no_emtpy_answers_embeddings_oversample_05:
    cmd: python src/etl/embeddings/transform/normalize_embeddings.py -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_embeddings/embeddings
      -n 50 --scatter_dataset -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_embeddings/embeddings
    - src/etl/embeddings/transform/normalize_embeddings.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings
  merge_norm_with_lengths:
    cmd: python src/etl/embeddings/transform/merge_embeddings_and_lengths.py -e /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings
      -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_data.pkl
      -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings_with_lengths
      --scatter_dataset
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_data.pkl
    - src/etl/embeddings/transform/merge_embeddings_and_lengths.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/normalized_embeddings_with_lengths
  merge_norm_with_lengths_oversample_05:
    cmd: python src/etl/embeddings/transform/merge_embeddings_and_lengths.py -e /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings
      -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_oversample_data.pkl
      -o /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings_with_lengths
      --scatter_dataset
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/train_text_lengths_oversample_data.pkl
    - src/etl/embeddings/transform/merge_embeddings_and_lengths.py
    outs:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2/oversample_normalized_embeddings_with_lengths
  classification:
    cmd: python src/etl/embeddings/classify/classification.py -d /data/gblanco/datasets/embeddings/quail_no_empty_answers_2
      -o /data/gblanco/results/quail_no_empty_answers_2 --metrics_dir data/metrics/classification
      --train --eval
    deps:
    - /data/gblanco/datasets/embeddings/quail_no_empty_answers_2
    - src/etl/embeddings/classify/classification.py
    params:
    - classification
    - features
    outs:
    - /data/gblanco/results/quail_no_empty_answers_2:
        checkpoint: true
    metrics:
    - data/metrics/classification/scores.json:
        cache: false
```

I hope this information is useful enough, but, if anything is needed just let me know!

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fixes #5451 

The change adds new flags: `--deps` for `dvc params diff` and `--param-deps` for `dvc exp show/diff` in order to show only those parameters that are actually used. It does not change current behaviour. 

Related: 
https://github.com/iterative/enhancement-proposals/pull/1

### CODE REVIEW

File: tests/func/params/test_diff.py [hunk 1]
```diff
@@ -194,19 +194,6 @@ def test_diff_deleted(tmp_dir, scm, dvc):
...
-    scm.add(["params.yaml", "Dvcfile"])
-    scm.commit("bar")
-
-    (tmp_dir / "params.yaml").unlink()
-
-    assert dvc.params.diff() == {
-        "params.yaml": {"foo": {"old": "bar", "new": None}}
-    }
-
-
```
Comment (pared) [inline tests/func/params/test_diff.py:209]: Copy of previous test (`test_diff_deleted`).

File: dvc/command/experiments.py [hunk 2]
```diff
@@ -783,6 +784,13 @@ def add_parser(subparsers, parent_parser):
         help="Exclude the specified params from output table.",
         metavar="<params_list>",
     )
+    experiments_show_parser.add_argument(
+        "-t",
+        "--tracked-params",
```
Comment (jorgeorpinel) [inline dvc/command/experiments.py:789]: They're all tracked (some by Git). What about `--deps` or `--only-deps` (or `--param-deps` in this case)

Context: https://github.com/iterative/dvc/issues/5451#issuecomment-794894902

File: dvc/command/experiments.py [hunk 3]
```diff
@@ -788,6 +790,12 @@ def add_parser(subparsers, parent_parser):
         help="Exclude the specified params from output table.",
         metavar="<params_list>",
     )
+    experiments_show_parser.add_argument(
+        "--param-deps",
+        action="store_true",
+        default=False,
+        help="Show only those params that are dependencies of stages.",
```
Comment (jorgeorpinel) [inline dvc/command/experiments.py:797]:
```suggestion
        help="Show only params that are stage dependencies.",
```

File: dvc/command/experiments.py [hunk 4]
```diff
@@ -873,6 +881,12 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Show unchanged metrics/params as well.",
     )
+    experiments_diff_parser.add_argument(
+        "--param-deps",
+        action="store_true",
+        default=False,
+        help="Show only those params that are dependencies of stages.",
```
Comment (jorgeorpinel) [inline dvc/command/experiments.py:888]:
```suggestion
        help="Show only params that are stage dependencies.",
```

File: dvc/command/params.py [hunk 5]
```diff
@@ -111,6 +112,12 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Show unchanged params as well.",
     )
+    params_diff_parser.add_argument(
+        "--deps",
+        action="store_true",
+        default=False,
+        help="Show only those params that are dependencies of stages.",
```
Comment (jorgeorpinel) [inline dvc/command/params.py:119]:
```suggestion
        help="Show only params that are stage dependencies.",
```

**Review by jorgeorpinel — COMMENTED** [review]
Names LGTM. One copy edit below (repeated 3 times).

I'm a bit surprised about the amount of logic needed to achieve this at first sight. Would it be easier to make this the default behavior and add option/logic to expand the scope? Well... I guess we have an entire discussion about that somewhere so never mind.

**Comment by pmrowla** [comment]
LGTM

**Review by pmrowla — APPROVED** [review]

**Pull request merged**